codegen/lco/generator.py

```diff
@@ -84,6 +93,8 @@ def generate_instrument_configs(ins_s: str, facility: str) -> str:
                     k.rstrip("s"): v
                     for k, v in ins["optical_elements"].items()
                 },
+                "configuration_extra_params": ins['validation_schema'].get('extra_params', {}).get('schema', {}),
+                "instrument_config_extra_params": ins['validation_schema'].get('instrument_configs', {}).get('schema', {}).get('schema', {}).get('extra_params', {}).get('schema', {})
```

Comment:
I really don't like how brittle this is. I know you think I was over-engineering it, but I was hoping for a more generic ability to iterate over fields and check for the existence of validate_schema for an object, not just hardcoding the json path for a specific instance of it. But maybe this is how it has to be done.

Reply:
The validation_schema on the instrument_type is applied at the `configuration` level of our request language. What you call brittle here is just me either returning the proper extra_params fields added for configuration or instrument_configuration (the most common ones used), or returning {} if none exists in the validation_schema. I'm not sure how that is brittle?